File: CPP/125.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        // std::isalnum(ch) -> if alphabet
        // std::tolower(ch) -> lower the ch
        int lt = 0, rt = s.size() - 1;

        // Do not use == in case the length is odd.
        while (lt < rt){
            if (!std::isalnum(s[lt])) {lt++; continue;}
            if (!std::isalnum(s[rt])) {rt--; continue;}

            if (std::tolower(s[lt]) != std::tolower(s[rt])) return false;
            else {
                lt++; rt--;
            }
        }
        return true;
    }
};

File: Python/125.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        lt = 0; rt = len(s) - 1;

        while (lt < rt):
            if not s[lt].isalnum():
                lt += 1; continue;
            if not s[rt].isalnum():
                rt -= 1; continue;

            # both lt and rt are alphabets
            if (s[lt].lower() != s[rt].lower()):
                return False
            else:
                lt += 1; rt -= 1;

        return True
